src/kontrol/prove.py

```diff
@@ -756,8 +756,19 @@ def _method_to_cfg(
                 f'Initial state proof {setup_proof.id} for {contract.name_with_path}.{method.signature} has no passing branches to build on. Method will not be executed.'
             )
 
+        # When minimizing constraints, we need to make sure not to forget any variables
+        # that might have been instantiated by a branching in the setup KCFG
+        keep_vars: set[str] = set().union(
+            *[
+                free_vars(constraint)
+                for split in cfg.splits()
+                for _, csubst in split.splits.items()
+                for constraint in csubst.constraints
+            ]
+        )
```

Comment:
Can you try this instead? It should create the set directly instead of creating a list of sets and then unpacking them.
```suggestion
        keep_vars: set[str] = {
            var
            for split in cfg.splits()
            for _, csubst in split.splits.items()
            for constraint in csubst.constraints
            for var in free_vars(constraint)
        }
```

Reply:
Yes, this works (if CI passes).